### DISCUSSION TITLE
i3-save-tree - doesn't work for some reason

### DISCUSSION THREAD

**Discussion by KleIKKK** [i3/i3#4603] [Q&A] [answered]
Hey guys :)
Just wanted to say I really love i3wm and I have been using it for quite some time, but lately after installing it on my new machine i cannot use command `i3-save-tree`. I am getting below error:

`Can't locate AnyEvent/I3.pm in @INC (you may need to install the AnyEvent::I3 module) (@INC contains: /usr/lib/perl5/5.34/site_perl /usr/share/perl5/site_perl /usr/lib/perl5/5.34/vendor_perl /usr/share/perl5/vendor_perl /usr/lib/perl5/5.34/core_perl /usr/share/perl5/core_perl) at /usr/bin/i3-save-tree line 19.
BEGIN failed--compilation aborted at /usr/bin/i3-save-tree line 19.`

Tried to google how to install it, but no luck. Any help would be really helpful.
Thank you and have a good one.

**Comment by Airblader** [comment, marked as answer]
[`AnyEvent-I3`](https://metacpan.org/dist/AnyEvent-I3) is a Perl module on which the script depends. If you installed i3 through your package manager, that dependency should've been installed, or at least be an optional dependency. If it is not, I would open raise an issue to the corresponding package maintainer(s).

How to install it manually (if you want to do so) depends a bit on your distribution and preferences. The generic way would be using cpan(m).

**Reply by KleIKKK** [reply]
@Airblader so let's assume it didn't install properly, and I got Arch Linux, any chance you could write how to install it?
Step by step like I am 10 years old if possible xD

**Reply by Airblader** [reply]
On Arch there is a package for it, [`perl-anyevent-i3`](https://archlinux.org/packages/community/any/perl-anyevent-i3/). Both i3-wm and i3-gaps depend on this.

**Reply by KleIKKK** [reply]
And now it works :) thank you very much
